src/types.rs

```diff
@@ -255,9 +255,12 @@ pub enum StrafeType {
 #[cfg_attr(feature = "proptest1", derive(Arbitrary))]
 pub enum StrafeDir {
     /// Turn left.
-    Left,
+    /// Data inside side strafe enum specifies constant yawspeed it should use
+    /// It should be optional for backward compatibility and empty field
+    /// means unlimited yawspeed. 0f32 is equivalent to empty field.
+    Left(Option<f32>),
     /// Turn right.
-    Right,
+    Right(Option<f32>),
```

Comment:
So we probably already discussed this at some point, but I don't entirely like putting this into the existing left and right strafing direction. Logically HLTAS strafing has a strafe type (first digit) and a strafe dir (second digit) and they can be combined together in most combinations, so you can e.g. wiggle strafe with max angle or const speed or max accel, or you can max angle strafe left, right, towards yaw, etc. Since constant yaw change effectively overrides any strafe type, I think it should instead be its own strafe type, which can only be used with left or right strafe dir. So you'd have s50 and s51 with the yaw set in the yaw field as now, and API-wise you'll have, say, `StrafeType::ConstTurn { turn_rate: f32 }` and the strafe dirs are left as is. Does that make sense?

Reply:
Right. I got your permission.